_topics/en/proof-of-reserves.md

```diff
@@ -0,0 +1,103 @@
+
+  - title: "BIP322 generic signmessage updated to use virtual transactions that support basic proof of reserves"
+    url: /en/newsletters/2020/10/07/#alternative-to-bip322-generic-signmessage
+
+  - title: "Kraken announces proof of reserves scheme"
+    url: /en/newsletters/2022/02/16/#kraken-announces-proof-of-reserves-scheme
+
+## Optional.  Same format as "primary_sources" above
+see_also:
+  - title: BIP322 signature proofs
+    link: BIP322
+
+## Optional.  Force the display (true) or non-display (false) of stub
+## topic notice.  Default is to display if the page.content is below a
+## threshold word count
+#stub: false
+
+## Required.  Use Markdown formatting.  Only one paragraph.  No links allowed.
+## Should be less than 500 characters
+excerpt: >
+  **Proof of reserves** are a timestamped commitment to a distribution
+  of funds signed by the entity who controls those funds.
+
+---
+For example, Alice, Bob, and Carol all have bitcoin-denominated deposits
+with Bank Corp.  Each week, Bank Corp uses the private key controlling
+their onchain funds to sign a message saying "Alice has 1 BTC on
+deposit; Bob has 2 BTC on deposit; and Carol has 3 BTC on deposit."  If
+the key Bank Corp uses to sign that message contains 6 BTC or more, the
+depositors have some assurance that Bank Corp could return their
+deposited amounts if it was willing.
+
+Proof of reserves can not guarantee that the funds are available.  For
+example, Bank Corp could contract with a third party who controls 6 BTC
+to have them generate the proof.  In that case, Bank Corp may not
+control any bitcoins even though its "proof" seems to indicate that it
+does.
+
+The simplest form of proof of reserves lists each depositor's name in a
+form that can't be confused with any other depositor's name.  Explicit
+names prevents the custodian from using the same bitcoins as proof for
+two or more different depositors.  For example, imagine two different
+people both named Alice who both have 1 BTC on deposit.  The example
+proof described for 6 BTC earlier would satisfy both of them, plus Bob
+and Carol, even though the actual total on deposit needs to be 7 BTC.
+
+For privacy, each depositor's name may be replaced with a pseudonym.
+However, depositors still need to ensure they are each given a unique
+name.  For more privacy, the association between name and exact deposit
+amount may be obscured, for example using a merkle-sum tree to show each
+depositor their specific amount and how it contributes towards the total
+amount of proven reserves.  For example, Alice is shown that her
+contribution is 1 BTC and that the contribution of the amount paired
+with hers is 2 BTC.  That aggregated amount is paired with some number of
+accounts also totaling 3 BTC for a grand total of 6 BTC:
+
+                   6 BTC
+                    / \
+              3 BTC     3 BTC
+             /     \
+    (1 BTC: Alice)
```

Comment:
I'm not sure, but does the node (hash) immediately to the right of Alice's node also need to be revealed? That is, both of the contributors to the "3 BTC" on the left side?

Reply:
Updated to separate the sum operations from the name operations, so it's (maybe) a little clearer that all Alice needs is the clear-text amount of the side branch and the hash commitment to the name of the person with that balance.